core/src/main/java/org/apache/brooklyn/core/mgmt/persist/BrooklynMementoPersisterToObjectStore.java

```diff
@@ -581,9 +583,16 @@ private Stopwatch deltaImpl(Delta delta, PersistenceExceptionHandler exceptionHa
             Stopwatch stopwatch = Stopwatch.createStarted();
             List<ListenableFuture<?>> futures = Lists.newArrayList();
             
+            Set<String> deletedIds = MutableSet.of();
```

Comment:
I'm surprised we need this. I can see that if one uses `ImmediateDeltaChangeListener` then it would happen - but that class is deprecated and not used (we should delete it).

If using `PeriodicDeltaChangeListener.persistNowInternal()` (which I thought all code paths would do), then it has already gone through this to remove the objects whose ids were removed.